app/models/lookup_keys/puppetclass_lookup_key.rb

```diff
@@ -41,6 +41,10 @@ def puppet?
   def check_override_selected
     return if (changed - ['description', 'override']).empty?
     return if override?
+    return if !override? && override_changed? && (changed - ['merge_overrides',
```

Comment:
A migration would be good, to keep this from continuing to run each time you save the lookup key.